app/helpers/sessions_helper.rb

```diff
@@ -14,6 +14,13 @@ def logged_in?
     !current_user.nil?
   end
 
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = 'Please log in first'
+      redirect_to login_url
+    end
+  end
+
   def log_out
     session.delete(:user_id)
     @current_user = nil
```

Comment:
Remove these methods from the view helpers as requested in #4. Decide on which controller should these methods reside. If you want to share them across controllers you can define them in the application controller.

Reply:
Hi Alvaro. I don't think we need to move those methods to the controller when we made the Sample App in the Rails Tutorial book the author put those methods in the `session_helper` too. Also if we put those methods in the controller we are breaking the convention to put actions that control the communication between models and views. Put them in the application controller is an option too but I think that the use of the helper is not a bad idea.
What do you think about that?